### PR TITLE
[Moloco-RMP] Rename RMP to MCM, Restoring test cases

### PR DESCRIPTION
**Pull request by heonjang**
## Testing

- [✅ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [✅ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)

## Description

1. Moloco RMP changed its name to Moloco MCM. Change the names in the code accordingly.
2. Restore removed breaking changes

- https://github.com/segmentio/action-destinations/commit/96ef3001e634001398a06b8894882aa20c3e8420
Screenshot 2024-03-26 at 4 09 42 PM
- https://github.com/segmentio/action-destinations/commit/25e96ae139b257e0ff236cc5b0ffd47df4e255eb
Screenshot 2024-03-26 at 4 09 16 PM

> TypeError: Invalid URL

However, if this was indeed a valid error, all other tests should have failed which was not the case.
Also same exact test did not fail when retried, so I assume those were some kinds of bug at the moment.
Screenshot 2024-03-26 at 4 12 11 PM
Screenshot 2024-03-26 at 4 13 06 PM

### CODE REVIEW

**Comment by heonjang** [comment]
@joe-ayoub-segment Hi Joe, it took a while to come back to you. Please take a look at the changes when you are available. Thank you in advance!

File: packages/destination-actions/src/destinations/moloco-rmp/__tests__/snapshot.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
```
Comment (heonjang) [inline packages/destination-actions/src/destinations/moloco-rmp/__tests__/snapshot.test.ts:1]: Restored snapshot test

File: packages/destination-actions/src/destinations/moloco-rmp/addToCart/__tests__/__snapshots__/snapshot.test.ts.snap [hunk 2]
```diff
@@ -1,51 +1,51 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`Testing snapshot for MolocoRmp's addToCart destination action: all fields 1`] = `
+exports[`Testing snapshot for MolocoMCM's addToCart destination action: all fields 1`] = `
 Object {
-  "channel_type": "2Tz2Ek0OF",
+  "channel_type": "uast)d8y]fWK6e",
```
Comment (heonjang) [inline packages/destination-actions/src/destinations/moloco-rmp/addToCart/__tests__/__snapshots__/snapshot.test.ts.snap:5]: Values changed due to the change in the `destination-slug` which is used for hashing values

File: packages/destination-actions/src/destinations/moloco-rmp/addToWishlist/__tests__/snapshot.test.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
```
Comment (heonjang) [inline packages/destination-actions/src/destinations/moloco-rmp/addToWishlist/__tests__/snapshot.test.ts:1]: Restored snapshot test

File: packages/destination-actions/src/destinations/moloco-rmp/index.ts [hunk 4]
```diff
@@ -19,10 +19,10 @@ import purchase from './purchase'
 import addToCart from './addToCart'
 
 const destination: DestinationDefinition<Settings> = {
-  name: 'Moloco Rmp',
-  slug: 'actions-moloco-rmp',
+  name: 'Moloco MCM',
+  slug: 'actions-moloco-mcm',
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/moloco-rmp/index.ts:23]: hi @heonjang - we can't change the slug. However the name change is OK. The slug isn't visible to customers other than in the URL.

**Review by joe-ayoub-segment — CHANGES_REQUESTED** [review]

**Comment by heonjang** [comment]
> Thanks @heonjang - looks good apart from the slug renaming which we can't support. Please see comment. Please let me know if that's OK?

Hi @joe-ayoub-segment, that is fine. I've reverted that change!

**Comment by joe-ayoub-segment** [comment]
hi @heonjang - looks like there are a couple of failing tests for your integration. Could you address these please?

Also there's a failing test because of another Integration - if you rebase from main this should be fixed. 

Best regards,
Joe

**Comment by heonjang** [comment]
Hi @joe-ayoub-segment,

I've updated the branch up to date with the `main`.

Also, I checked that both

1. unit tests
Screenshot 2024-04-04 at 8 19 07 PM

2. local server tests
Screenshot 2024-04-04 at 8 18 05 PM

passed.

It seems like CI can be run after approval from you. Please take a look whenever you are available!

**Comment by joe-ayoub-segment** [comment]
hi @heonjang - I've rerun CI checks and there still seems to be some sort of issue. 
Here's a screenshot from the log: 
image

File: packages/destination-actions/src/destinations/moloco-rmp/__tests__/snapshot.test.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+const testDestination = createTestIntegration(destination)
+const destinationSlug = 'actions-moloco-rmp'
+
+describe(`Testing snapshot for ${destinationSlug} destination:`, () => {
+  for (const actionSlug in destination.actions) {
+    it(`${actionSlug} action - required fields`, async () => {
+      const seedName = `${destinationSlug}#${actionSlug}`
+      const action = destination.actions[actionSlug]
+      const [eventData, _] = generateTestData(seedName, destination, action, true)
+      const settingsData = {
```
Comment (heonjang) [inline packages/destination-actions/src/destinations/moloco-rmp/__tests__/snapshot.test.ts:15]: Use a constant for settingsData

**Comment by heonjang** [comment]
Hi @joe-ayoub-segment 

Somehow, the snapshot test succeeded in a local environment but failed in the Segment's CI.

image

However, this error doesn't make sense because other actions (HOME, PURCHASE, ...) did not fail even though they share the same logic.

One possibility is that when creating a `settingsData`, the test code creates mock data by hashing the action. So somehow the LAND's result contained an invalid `platform` value which is used to create a target URL.
```
const [eventData, settingsData] = generateTestData(seedName, destination, action, true)
```

To avoid this issue, I've updated the `settingsData` so that it always contains valid data.

**Comment by heonjang** [comment]
> Hi @joe-ayoub-segment
> 
> Somehow, the snapshot test succeeded in a local environment but failed in the Segment's CI.
> 
> image
> 
> However, this error doesn't make sense because other actions (HOME, PURCHASE, ...) did not fail even though they share the same logic.
> 
> One possibility is that when creating a `settingsData`, the test code creates mock data by hashing the action. So somehow the LAND's result contained an invalid `platform` value which is used to create a target URL.
> 
> ```
> const [eventData, settingsData] = generateTestData(seedName, destination, action, true)
> ```
> 
> To avoid this issue, I've updated the `settingsData` so that it always contains valid data. ([#1953 (comment)](https://github.com/segmentio/action-destinations/pull/1953#discussion_r1556657672))

Actually,  I just removed the problematic test files so that this PR only focuses on the renaming. Let me handle the test cases separately :(

**Comment by joe-ayoub-segment** [comment]
Hi @heonjang - the CI is still failing but I don't think it has anything to do with your code - so I'm happy to merge and deploy it next Tuesday.

**Comment by joe-ayoub-segment** [comment]
Hi @heonjang - PR deployed. I did need to remove some tests again as they were causing issues with our CI!

**Comment by joe-ayoub-segment** [comment]
Please confirm that you are happy with the change.

**Comment by heonjang** [comment]
@joe-ayoub-segment Hi Joe, sorry for missing the last comment.

I checked whether the integration's name was changed or not, but it seems like the destination's name has not been updated yet.

image

Do you have an idea on this?

**Comment by joe-ayoub-segment** [comment]
hi @heonjang that's been fixed now. 

Best regards,
Joe

**Comment by heonjang** [comment]
> hi @heonjang that's been fixed now.
> 
> Best regards, Joe

Thanks! Confirmed that it's been fixed. Have a great rest of your week :)

**Pull request merged**